### ISSUE TITLE
rst2html5 download fails

### ISSUE THREAD

**Issue by sumnerevans** [NixOS/nixpkgs#106268]
**Describe the bug**
Due to a recent change in 90aacae8a503cbf592d8b422b83f02e9bd6cb274, the `rst2html5` package no longer builds. This is the error:

```
$ nix-env -iA nixos.rst2html5
installing 'python3.8-rst2html5-1.10.6'
these derivations will be built:
  /nix/store/vn2jdpppcrmw6q1qsq94cs1ch46gcc3z-rst2html5-1.10.6.tar.gz.drv
  /nix/store/nw2rf9p58idy04d6wsa1cay67r649514-python3.8-rst2html5-1.10.6.drv
building '/nix/store/vn2jdpppcrmw6q1qsq94cs1ch46gcc3z-rst2html5-1.10.6.tar.gz.drv'...

trying https://files.pythonhosted.org/packages/source/r/rst2html5/rst2html5-1.10.6.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404

trying https://pypi.io/packages/source/r/rst2html5/rst2html5-1.10.6.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   122  100   122    0     0   2975      0 --:--:-- --:--:-- --:--:--  2975
100   279  100   279    0     0   2113      0 --:--:-- --:--:-- --:--:--  2113
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
error: cannot download rst2html5-1.10.6.tar.gz from any mirror
builder for '/nix/store/vn2jdpppcrmw6q1qsq94cs1ch46gcc3z-rst2html5-1.10.6.tar.gz.drv' failed with exit code 1
cannot build derivation '/nix/store/nw2rf9p58idy04d6wsa1cay67r649514-python3.8-rst2html5-1.10.6.drv': 1 dependencies couldn't be built
error: build of '/nix/store/nw2rf9p58idy04d6wsa1cay67r649514-python3.8-rst2html5-1.10.6.drv' failed
```

**To Reproduce**
1. Run `nix-env -iA nixos.rst2html5`

**Expected behavior**
It should build

**Additional context**
The reason for this error is that there is no `tar.gz` published on PyPi (see here: https://pypi.org/project/rst2html5/#files). I'm not sure where to go to actually get the source code, either.

**Notify maintainers**
@AndersonTorres 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.9.12, NixOS, 21.03pre257339.83cbad92d73 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.9`
 - channels(sumner): `"home-manager"`
 - channels(root): `"nixos-21.03pre257339.83cbad92d73, nixpkgs-21.03pre251181.dd1b7e377f6"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  rst2html5
# a list of nixos modules affected by the problem
module:
```

**Comment by AndersonTorres** [comment]
Why does it not download the Wheel file instead of a generic TGZ?

**Comment by AndersonTorres** [comment]
Already fixed via #114536
